### PR TITLE
[7354] Implement ITT reform on Register API - part 1

### PR DESCRIPTION
**Pull request by Reyko**
### Context

[7354-implement-itt-reform-on-register-api](https://trello.com/c/gMzDEEyF/7354-implement-itt-reform-on-register-api)

As part of the ITT reform the API has been refactored to receive and return references to `lead_partner` instead of `lead_school`

To maintain compatibility with the application, lead schools will use the `lead_partner_not_applicable` column until we fully migrate to lead partners.

### Changes proposed in this pull request

* Rename `lead_school_urn` to `lead_partner_urn` in the API
* Rename `lead_school_not_applicable` to `lead_partner_not_applicable` in the API and db
* Update docs

### Important business

- [ ] Does this PR introduce any PII fields that need to be overwritten or deleted in db/scripts/sanitise.sql?
- [x] Does this PR change the database schema? If so, have you updated the config/analytics.yml file and considered whether you need to send 'import_entity' events?
- [ ] Do we need to send any updates to DQT as part of the work in this PR?
- [ ] Does this PR need an ADR?

NB: Please notify the #twd_data_insights team and ask for a review if new fields are being added to analytics.yml

### CODE REVIEW

File: app/serializers/api/v0_1/trainee_serializer.rb [hunk 1]
```diff
@@ -183,11 +184,11 @@ def employing_school_urn
       end
 
       def lead_partner_urn_ukprn
-        @trainee.lead_school&.urn
+        @trainee.lead_partner&.ukprn
```
[the remark this answers is not in the document]
Comment (Reyko) [reply app/serializers/api/v0_1/trainee_serializer.rb:187]: Refactored https://github.com/DFE-Digital/register-trainee-teachers/pull/4485/commits/5ae0eb9b4070c4cf7c43e7e653964669e24cd9b7

File: app/services/api/v0_1/hesa_mapper/attributes.rb [hunk 2]
```diff
@@ -194,19 +194,19 @@ def funding_entity_id
...
-        def school_attributes
+        def lead_partner_and_employing_school_attributes
           attrs = {}
 
-          return attrs if params[:lead_school_urn].blank?
+          return attrs if params[:lead_partner_urn].blank?
 
-          if NOT_APPLICABLE_SCHOOL_URNS.include?(params[:lead_school_urn])
-            attrs.merge!(lead_school_not_applicable: true)
+          if NOT_APPLICABLE_SCHOOL_URNS.include?(params[:lead_partner_urn])
```
Comment (adamjcolvin) [inline app/services/api/v0_1/hesa_mapper/attributes.rb:202]: Small thing, but should this constant now be `NOT_APPLICABLE_LEAD_PARTNER_URNS` or something like that?

File: app/views/trainees/lead_schools/_results.html.erb
Comment (adamjcolvin) [inline app/views/trainees/lead_schools/_results.html.erb:1]: Should we be renaming files as part of this PR? Should this now be `trainees/lead_partners/_results.html.erb?`

File: app/views/trainees/lead_schools/edit.html.erb
Comment (adamjcolvin) [inline app/views/trainees/lead_schools/edit.html.erb:1]: Same question here, should this file be renamed to `trainees/lead_partners/edit.html.erb`?

File: spec/features/form_sections/teacher_training_data/edit_schools_spec.rb [hunk 3]
```diff
@@ -24,7 +24,7 @@
     scenario "choosing not applicable for lead school" do
       i_click_on_change_school(:lead_school)
       and_i_am_on_the_edit_lead_school_page
```
Comment (adamjcolvin) [inline spec/features/form_sections/teacher_training_data/edit_schools_spec.rb:26]: References to lead school here, and in a few other methods used in this spec.

File: spec/features/form_sections/teacher_training_data/edit_schools_spec.rb
Comment (adamjcolvin) [inline spec/features/form_sections/teacher_training_data/edit_schools_spec.rb:1]: Should this now be `edit_lead_partners_spec.rb`?

**Review by adamjcolvin — COMMENTED** [review]
@Reyko looks good, I'm just not sure how far we should be going with the renaming of things at this point, so sorry if some of the suggestions aren't relevant. We still have some views and partials under `lead_schools` for example, and I don't know if they need changing. As well as some helper method in specs and a few other places. Let me know what you think.

**Review by adamjcolvin — APPROVED** [review]

**Comment by Reyko** [comment]
> this ticket

I suggest to coordinate with @kennyevil as he is implementing changes similar to your ticket 

- [7177-update-existing-provider-led-routes-so-that-providers-can-optionally-add-a-lead-partner](https://trello.com/c/VFds9crk/7177-update-existing-provider-led-routes-so-that-providers-can-optionally-add-a-lead-partner)
- https://github.com/DFE-Digital/register-trainee-teachers/pull/4478

**Pull request merged**